File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int length_ll(ListNode* head)
    {
        int count_node=0;
        ListNode *ptr=head;
        while(ptr!=NULL)
        {
            count_node++;
            ptr=ptr->next;
        }
        return count_node;
    }
    ListNode* removeNthFromEnd(ListNode* head, int n) 
    {
        int len=length_ll(head);
        if(len==1)
            return NULL;
        if(len==n)
            return head->next;
        int node_number=0;
        ListNode *ptr=head;
        while( node_number < len-n-1 )
        {
            ptr=ptr->next;
            node_number++;
        }
        ptr->next=ptr->next->next;
        return head;
    }
};

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        if(head==null){
            return null;
        }
        ListNode curr= head;
        int size = 0;
        
        while(curr!=null){
            size++;
            curr=curr.next;
        }
        if(size==n){
            return head.next;
        }
        int it= size-n;
        
        ListNode curr2= head;
        int i =1;
        while(i<it){
            curr2=curr2.next;
            i++;
        }
        curr2.next = curr2.next.next;
        return head;
    }
}